internal/bminventory/inventory.go

```diff
@@ -4516,6 +4517,14 @@ func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installe
 		c = &cluster.Cluster
 	}
 
+	// When the assisted service is running on the same host as the agent, this
+	// host must become the bootstrap so that it installs last
+	if b.InEphemeralInstaller && hostutil.IsAssistedServiceHost(host) {
+		log.Infof("host <%s> also hosts assisted-service", params.NewHostParams.HostID.String())
+		host.Role = models.HostRoleMaster
+		host.Bootstrap = true
+	}
```

Comment:
Could we get here some handling for a scenario where more than one host tries to become the bootstrap? I can see that this is determined by the existence of a file in a local filesystem (`/var/run/assisted-agent/host-id`), so this looks like a potential for 2 hosts in the infra to get in a race. Or not? If not, then maybe a comment in the function below `IsAssistedServiceHost()`